src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -313,6 +319,31 @@ async Task<MsalAccessTokenCacheItem> ITokenCacheInternal.FindAccessTokenAsync(Au
             return null;
         }
 
+        private MsalAccessTokenCacheItem FilterATByKeyId(MsalAccessTokenCacheItem item, AuthenticationRequestParameters authenticationRequest)
+        {
+            if (item == null)
+            {
+                return null;
+            }
+
+            string requestKid = authenticationRequest.AuthenticationScheme.KeyId;
+            if (string.IsNullOrEmpty(item.Kid) && string.IsNullOrEmpty(requestKid))
+            {
```

Comment:
is that the only way to identify a bearer token?

Reply:
This is a good point. 

I think I will add an explicit "TokenType" in the cache. After all, this is returned by AAD in the token response. 

For example, AAD will have the ability to issue SSH "tokens", which are also bound to keys.